include/winrospublisher.h

```diff
@@ -108,7 +108,7 @@ namespace ros_msft_camera
                         ros_image->step = -Stride;
                         size_t size = ros_image->step * u32Height;
                         ros_image->data.resize(size);
-                        for (int i = 0; i < u32Height; i++)
+                        for (uint32_t i = 0; i < u32Height; i++)
                         {
                             memcpy(ros_image->data.data(), pix + Stride * i, -Stride);
```

Comment:
I might have left a bug here. The destination pointer is not getting updated in loop iterations.
Should be something like ros_image->data.data() - Stride*i

Reply:
I'm going to test this - I wonder if that's why I was seeing the false color image?

Reply:
I doubt this is related to your bug. This bug will cause only on line of active pixels in the output, unlike the stretched color distorted image we saw for your bug. This bug will show up only if the stride in captured buffer is negative. This rarely happens with newer cameras.